pkg/mapper/configmap/configmap.go

```diff
@@ -111,31 +116,77 @@ func (err ErrParsingMap) Error() string {
 	errs := make([]error, 0)
+	rawUserMappings := make([]config.UserMapping, 0)
 	userMappings = make([]config.UserMapping, 0)
+	userArnLikeMappings = make([]config.UserMapping, 0)
 	if userData, ok := m["mapUsers"]; ok {
 		userJson, err := utilyaml.ToJSON([]byte(userData))
 		if err != nil {
 			errs = append(errs, err)
 		} else {
-			err = json.Unmarshal(userJson, &userMappings)
+			err = json.Unmarshal(userJson, &rawUserMappings)
 			if err != nil {
 				errs = append(errs, err)
 			}
+
+			for _, userMapping := range rawUserMappings {
+				switch {
+				case userMapping.UserARN == "" && userMapping.UserARNLike == "":
+					errs = append(errs, fmt.Errorf("One of userarn or userarnLike must be supplied"))
+				case userMapping.UserARN != "" && userMapping.UserARNLike != "":
+					errs = append(errs, fmt.Errorf("Only one of userarn or userarnLike can be supplied"))
+				case userMapping.UserARN != "":
+					userMappings = append(userMappings, userMapping)
+				case userMapping.UserARNLike != "":
+					ok, err := arnlike.ArnLike(userMapping.UserARNLike, "arn:*:iam:*:*:user/*")
+					if err != nil {
+						errs = append(errs, err)
+					} else if !ok {
+						errs = append(errs, fmt.Errorf("UserARNLike '%s' did not match an ARN for an IAM User", userMapping.UserARNLike))
+					}
+					userArnLikeMappings = append(userArnLikeMappings, userMapping)
+				default:
+					errs = append(errs, fmt.Errorf("Unexpected error parsing userMapping: %v", userMapping))
+				}
+			}
 		}
 	}
 
+	rawRoleMappings := make([]config.RoleMapping, 0)
 	roleMappings = make([]config.RoleMapping, 0)
+	roleArnLikeMappings = make([]config.RoleMapping, 0)
 	if roleData, ok := m["mapRoles"]; ok {
 		roleJson, err := utilyaml.ToJSON([]byte(roleData))
 		if err != nil {
 			errs = append(errs, err)
 		} else {
-			err = json.Unmarshal(roleJson, &roleMappings)
+			err = json.Unmarshal(roleJson, &rawRoleMappings)
 			if err != nil {
 				errs = append(errs, err)
 			}
+
+			for _, roleMapping := range rawRoleMappings {
+				switch {
+				case roleMapping.RoleARN == "" && roleMapping.RoleARNLike == "":
+					errs = append(errs, fmt.Errorf("One of rolearn or rolearnLike must be supplied"))
+				case roleMapping.RoleARN != "" && roleMapping.RoleARNLike != "":
+					errs = append(errs, fmt.Errorf("Only one of rolearn or rolearnLike can be supplied"))
+				case roleMapping.RoleARN != "":
+					roleMappings = append(roleMappings, roleMapping)
+				case roleMapping.RoleARNLike != "":
+					ok, err := arnlike.ArnLike(roleMapping.RoleARNLike, "arn:*:iam:*:*:role/*")
+					if err != nil {
+						errs = append(errs, err)
+					} else if !ok {
+						errs = append(errs, fmt.Errorf("RoleARNLike '%s' did not match an ARN for an IAM Role", roleMapping.RoleARNLike))
+					}
+					roleArnLikeMappings = append(roleArnLikeMappings, roleMapping)
+				default:
+					errs = append(errs, fmt.Errorf("Unexpected error parsing roleMapping: %v", roleMapping))
+				}
+			}
```

Comment:
I'm confused why you need separate maps for `UserMapping` and `RoleMapping` for ArnLike or non-ArnLike matchers. You could probably simplify things quite a bit by creating to helper methods for `UserMapping` and `RoleMapping` called `Validate` and `Matches`:

```go
// Validate returns whether the RoleMapping is valid after being
// unmarshaled from JSON configuration
func (m *RoleMapping) Validate() error {
	if m ==  nil {
		return fmt.Errorf("RoleMapping was nil")
	}
	if m.RoleARN == "" && m.RoleARNLike == "" {
		return fmt.Errorf("One of rolearn or rolearnLike must be supplied"))
	} else if m.RoleARN != "" && m.RoleARNLike != "":
		return fmt.Errorf("Only one of rolearn or rolearnLike can be supplied"))
	}
	if m.RoleARNLike != "" {
		ok, err := arnlike.ArnLike(m.RoleARNLike, "arn:*:iam:*:*:role/*")
		if err != nil {
			return err
		} else if !ok {
			return fmt.Errorf("RoleARNLike '%s' did not match an ARN for an IAM Role", m.RoleARNLike)
		}
	}
	return nil
}

// Matches returns true if the supplied ARN or ARN-like thing matches
// this RoleMapping
func (m *RoleMapping) Matches(subject string) bool {
	if m.RoleARN != "" {
		return m.RoleARN == subject
	}
	ok, _ := arnlike.ArnLike(subject, m.RoleARNLike)
	return ok
}
```